cmd/libs/go2idl/parser/parse.go

```diff
@@ -127,21 +143,30 @@ func (b *Builder) buildPackage(pkgPath string) (*build.Package, error) {
 	return pkg, nil
 }
 
-// AddFile adds a file to the set. The name must be of the form canonical/pkg/path/file.go.
-func (b *Builder) AddFile(name string, src []byte) error {
-	return b.addFile(name, src, true)
+// AddFile adds a file to the set. The pkg must be of the form
+// "canonical/pkg/path" and the path must be the absolute path to the file. A
+func (b *Builder) AddFile(pkg string, name string, src []byte) error {
```

Comment:
done

On Mon, Apr 25, 2016 at 1:10 PM, Daniel Smith <notifications@github.com>
wrote:

> In cmd/libs/go2idl/parser/parse.go
> <https://github.com/kubernetes/kubernetes/pull/24242#discussion_r60979630>
> :
>
> > @@ -127,21 +143,30 @@ func (b *Builder) buildPackage(pkgPath string) (*build.Package, error) {
> >  	return pkg, nil
> >  }
> >
> > -// AddFile adds a file to the set. The name must be of the form canonical/pkg/path/file.go.
> > -func (b *Builder) AddFile(name string, src []byte) error {
> > -	return b.addFile(name, src, true)
> > +// AddFile adds a file to the set. The pkg must be of the form
> > +// "canonical/pkg/path" and the path must be the absolute path to the file. A
> > +func (b *Builder) AddFile(pkg string, name string, src []byte) error {
>
> s/name/path/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24242/files/c45dc75e44061a01d7b49fb7237b572d23b067a3..27902ce681296b1bbf11345fe1c83209fcba49e0#r60979630>
>